pkgs/stdenv/generic/make-derivation.nix

```diff
@@ -232,6 +232,8 @@ else let
 
   outputs = outputs';
 
+  referencesOnBuild = nativeBuildInputs ++ propagatedNativeBuildInputs;
+  referencesOnHost = buildInputs ++ propagatedBuildInputs;
   references = nativeBuildInputs ++ buildInputs
```

Comment:
`references` was added in https://github.com/nixos/nixpkgs/commit/eaee2a119916b2761d08cb8f68f1b4afbde15e51

> This option makes `meta.evaluate` into a close approximation of the result of
evaluating `.outPath` by checking all the dependencies recursively at a cost of
2x slowdown. Note that actually evaluating `.outPath` costs some
5x-7x more because `.outPath` also computes all the hashes.

This pr has big performance impacts too
![image](https://github.com/NixOS/nixpkgs/assets/56650223/0c2ba7ee-6a5a-4aa4-895c-0493b150c301)

Reply:
> This pr has big performance impacts too (current ofborg eval time is ~11 minutes)

Percentages are more useful here: ~5% on all but one metric (and that one metric at +10% is **not** an allocation metric).

That said, there are still lots of optimization opportunities.  If performance is the only remaining obstacle I will implement those.  I believe the overhead can be brought down to ~1% but I don't want to put in that effort if it's going to be wasted.

Reply:
1c68dedefcb5ac3c2b2757fc076a1f9850c4c1a1 should improve this.  Let's see what OfBorg says.

Reply:
Latest eval report, at 4141be0.  It's basically neutral; a mixed bag of under-5% changes (about half faster, half slower).  The only thing that stands out is the +12% on `nrLookups`, but attrset lookups are fast and allocate no memory -- this metric is far less important than most of the others.

![eval-report](https://github.com/NixOS/nixpkgs/assets/54836058/e38a5c67-631e-445d-aa7c-b98590206a9e)